float64 y_diff
---
bool spray_completed